src/notifications/TwitchNotifier.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const fetch = require('node-fetch');
+const moment = require('moment');
+const TwitchEmbed = require('../embeds/TwitchEmbed');
+const Broadcaster = require('./Broadcaster');
+const { platforms } = require('../CommonFunctions');
+const logger = require('../Logger');
+
+/**
+ * Watches for Twitch go-lives and broadcasts them
+ */
+class TwitchNotifier {
+  constructor({
+    client, settings, messageManager,
+  }) {
+    this.broadcaster = new Broadcaster({
+      client,
+      settings,
+      messageManager,
+    });
+
+    this.lastStartedAtTime = null;
+    this.user = process.env.TWITCH_USER_LOGIN || 'warframe';
+
+    // save this in closure for setInterval
+    const self = this;
+    setInterval(async () => { await self.pollTwitch(); },
+      process.env.TWITCH_POLL_INTERVAL_MS || 60000);
+  }
+
+  /**
+   * Retrieves the details of the twitch user that we are following
+   * @returns {Object} a twitch api user response object
+   */
+  async getUserDetails() {
+    try {
+      const response = await fetch(`https://api.twitch.tv/helix/users?login=${this.user}`, {
+        method: 'GET',
+        headers: {
+          'Client-ID': process.env.TWITCH_CLIENT_ID,
+        },
+      })
+        .then(data => data.json());
+
+      if (response.data.length > 0) {
+        return response.data[0];
+      }
+    } catch (error) {
+      logger.error(error);
+    }
+    return undefined;
+  }
+
+  /**
+   * Checks the user's stream to see if they are currently live
+   */
+  async pollTwitch() {
+    try {
+      const response = await fetch(`https://api.twitch.tv/helix/streams?user_login=${this.user}`, {
+        method: 'GET',
+        headers: {
+          'Client-ID': process.env.TWITCH_CLIENT_ID,
+        },
+      })
+        .then(data => data.json());
+
+      // If the stream query has some results, then the user is live
+      if (response.data.length > 0) {
+        // parse what time the stream started
+        const startedAt = moment(response.data[0].started_at);
+
+        // if we havent seen the stream start yet
+        // OR the stream started after the last time it started, notify
+        if (this.lastStartedAtTime == null || startedAt.isAfter(this.lastStartedAtTime)) {
+          // get the user details now so we can enhance the embed with profile data
+          const userDetails = await this.getUserDetails();
```

Comment:
is this something we could run once at startup, since we don't really to fetch it each time, right?

Reply:
We could, but I don’t know what will happen if they change their pfp. DE probably doesn’t change it often enough that it matters. I can do it in the ctor.

Reply:
I moved it to a start() function that I call right after it is constructed. async/await cannot be used from a constructor.